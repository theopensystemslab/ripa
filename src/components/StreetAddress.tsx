import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Container from "@material-ui/core/Container";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import { useFormik } from "formik";
import * as React from "react";

interface IText {
  title: string;
  name: string;
  type: string;
  options: string[];
}

export const StreetAddress: React.FC<IText> = ({
  title,
  type,
  name,
  options
}) => {
  const formik = useFormik({
    initialValues: {},
    onSubmit: values => {
      console.log(JSON.stringify(values, null, 2));
    }
  });
  const renderLabels = el => {
    switch (el) {
      case "building":
        return "Building and street";
      case "street":
        return "";
      case "city":
        return "Town or City";
      case "county":
      case "postcode":
        return el;
      default:
        break;
    }
  };
  return (
    <Container maxWidth="md">
      <Box bgcolor="background.paper" p={4} maxWidth={500}>
        <form onSubmit={formik.handleSubmit}>
          <Typography variant="h4" gutterBottom>
            <strong>{title}</strong>
          </Typography>
          {options.map((el, index) => (
            <div key={`${el}-${index}`}>
              <Box mb={2.5}>
                <TextField
                  required={el === "street" || el === "city"}
                  onChange={formik.handleChange}
                  placeholder={el}
                  label={renderLabels(el)}
                  fullWidth
                  type={type}
                  name={`${name}-${el}`}
                />
              </Box>
            </div>
          ))}
          <Box textAlign="right">
            <Button type="submit" variant="contained" color="primary">
              Look up address
            </Button>
          </Box>
        </form>
      </Box>
    </Container>
  );
};

export default StreetAddress;
