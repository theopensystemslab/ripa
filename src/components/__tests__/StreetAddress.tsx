import { cleanup, fireEvent, render, wait } from "@testing-library/react";
import * as React from "react";

import StreetAddress from "../StreetAddress";

afterEach(cleanup);

describe("Street Address Component", () => {
  it("should render snapshot", () => {
    const title = " test title";
    const name = "test name";
    const type = "text";
    const options = ["building", "street", "city", "county", "postcode"];
    const { asFragment } = render(
      <StreetAddress title={title} type={type} options={options} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
