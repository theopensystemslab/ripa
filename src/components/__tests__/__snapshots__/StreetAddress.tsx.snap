// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Street Address Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-270"
  >
    <form
      data-testid="streetAddressForm"
    >
      <div
        class="MuiBox-root MuiBox-root-271"
      >
        <div
          class="MuiTypography-root makeStyles-question-272 MuiTypography-h4"
        >
          <strong>
            <strong>
              test title
            </strong>
          </strong>
        </div>
      </div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-303"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Building and street
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="building"
                placeholder="building"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-361"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="street"
                placeholder="street"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-362"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Town or City
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="city"
                placeholder="city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-363"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              country
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="country"
                placeholder="country"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-364"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              postcode
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="postcode"
                placeholder="postcode"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-365"
      >
        <div />
        <div />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
